WireAnalytics/Sources/WireAnalytics/Service/AnalyticsService.swift

```diff
@@ -0,0 +1,219 @@
+        }
+
+        print("[ANALYTICS] enabling tracking")
+
+        let countly = countlyProvider()
+        self.countly = countly
+
+        countly.start(
+            appKey: config.secretKey,
+            host: config.serverHost
+        )
+    }
+
+    /// Stop sending analytics data.
+
+    public func disableTracking() throws {
+        guard let countly else {
+            throw AnalyticsServiceError.serviceIsNotConfigured
+        }
+
+        print("[ANALYTICS] disabling tracking")
+
+        countly.endSession()
+        clearCurrentUser()
+        countly.resetInstance()
+        self.countly = nil
+    }
+
+    // MARK: - User
+
+    /// Switch the current analytics user.
+    ///
+    /// - Parameter user: The user to switch to.
+
+    public func switchUser(_ user: AnalyticsUser) {
+        guard
+            let countly,
+            user != currentUser
+        else {
+            return
+        }
+
+        print("[ANALYTICS] switching user")
+
+        countly.endSession()
+
+        pushUser(
+            user,
+            mergeData: false
+        )
+
+        countly.beginSession()
+    }
+
+    /// Update the current user.
+    ///
+    /// If the user's id changed, all previously tracked data associated with
+    /// the current id will be associated with the new id.
+    ///
+    /// - Parameter user: The updated current user.
+
+    public func updateCurrentUser(_ user: AnalyticsUser) {
+        guard
+            let currentUser,
+            user != currentUser
+        else {
+            return
+        }
+
+        print("[ANALYTICS] updating current user")
+
+        pushUser(
+            user,
+            mergeData: true
+        )
+    }
+
+    private func pushUser(
+        _ user: AnalyticsUser?,
+        mergeData: Bool
+    ) {
+        guard let countly else {
+            return
+        }
+
+        if let id = user?.analyticsIdentifier {
+            countly.changeDeviceID(
+                id,
+                mergeData: mergeData
+            )
+        }
+
+        countly.setUserValue(
+            user?.teamInfo?.id,
+            forKey: AnalyticsUserKey.teamID.rawValue
+        )
+
+        countly.setUserValue(
+            user?.teamInfo?.role,
+            forKey: AnalyticsUserKey.teamRole.rawValue
+        )
+
+        countly.setUserValue(
+            user?.teamInfo.map { String($0.size.logRound()) },
+            forKey: AnalyticsUserKey.teamSize.rawValue
+        )
+
+        currentUser = user
+    }
+
+    private func clearCurrentUser() {
+        print("[ANALYTICS] clearing current user")
+
+        pushUser(
+            nil,
+            mergeData: false
+        )
+    }
+
+    // MARK: - Event
+
+    /// Track an event.
+    ///
+    /// - Parameter event: The event to track.
+
+    public func trackEvent(_ event: AnalyticsEvent) {
+        guard
+            let countly,
+            let currentUser
+        else {
+            return
```

Comment:
maybe we should throw here, or at least log

Reply:
Wouldn't we then pollute the logs when analytics is disabled or not configured for the build?

Reply:
added log: fc2950894ed9913112722e521e7ec53d5f7d96f4